server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/ShardsLimitAllocationDecider.java

```diff
@@ -70,11 +72,9 @@
         Property.NodeScope
     );
 
-    private final Settings settings;
-
     public ShardsLimitAllocationDecider(Settings settings, ClusterSettings clusterSettings) {
-        this.settings = settings;
         this.clusterShardLimit = CLUSTER_TOTAL_SHARDS_PER_NODE_SETTING.get(settings);
+        this.indexShardLimit = INDEX_TOTAL_SHARDS_PER_NODE_SETTING.get(settings);
```

Comment:
I think this is not necessary, this setting is not legal as a node setting.

Reply:
Right 🤦 removed this, making the rest of the code far simpler as well.